wordpress-mobile/WordPress-Android
# Replace event/observeEvent with Event wrapper for single actions

This is an exploration related to this article - https://medium.com/androiddevelopers/livedata-with-snackbar-navigation-and-other-events-the-singleliveevent-case-ac2622673150

I'm replacing a parent `Event` and the `observeEvent` method with an `Event` wrapper. I think this approach makes it less error prone. You can define an event on the observable point of code and the observer has to call the correct method to get the value. It adds a bit of boilerplate code.

What do you think about this solution?

To test:
* Go to stats
* Click on View more on any block
* Go back
* Click on View more on the same block
* Go to DWMY
* Change date from date selector
* Change date from the Overview block
* Check that all works as expected
